doc/src/build/move/sui-move-library.md

```diff
@@ -23,54 +23,43 @@ transfer::transfer(obj, recipient);
 
-A common pattern of object owning another object is to have a field in the parent object to track the ID of the child object. It is important to ensure that we keep such a field's value consistent with the actual ownership relationship. For example, we do not end up in a situation where the parent's child field contains an ID pointing to object A, while in fact the parent owns object B. To ensure the consistency, we defined a custom type called `ChildRef` to represent object ownership. Whenever an object is transferred to another object, a `ChildRef` instance is created to uniquely identify the ownership. The library implementation ensures that the `ChildRef` goes side-by-side with the child object so that we never lose track or mix up objects.
-To transfer an object `obj` (whose owner is an address) to another object `owner`:
-```
-transfer::transfer_to_object(obj, &mut owner);
-```
-This function returns a `ChildRef` instance that cannot be dropped arbitrarily. It can be stored in the parent as a field.
-Sometimes we need to set the child field of a parent while constructing it. In this case, we don't yet have a parent object to transfer into. In this case, we can call the `transfer_to_object_id` API. Example:
-```
-let parent_info = object::new(ctx);
-let child = Child { info: object::new(ctx) };
-let (parent_id, child_ref) = transfer::transfer_to_object_id(child, parent_info);
-let parent = Parent {
-    info: parent_info,
-    child: child_ref,
-};
-transfer::transfer(parent, tx_context::sender(ctx));
-```
-To transfer an object `child` from one parent object to a new parent object `new_parent`, we can use the following API:
 ```
-transfer::transfer_child_to_object(child, child_ref, &mut new_parent);
+use sui::dynamic_object_field as ofield;
+
+let a: &A = /* ... */;
+let b: &B = /* ... */;
+
+// Adds `b` as a dynamic object field to `a` with "name" `0: u8`.
+ofield::add<u8, B>(&mut a.id, 0, b);
 ```
-Note that in this call, we must also have the `child_ref` to prove the original ownership. The call will return a new instance of `ChildRef` that the new parent can maintain.
-To transfer an object `child` from an object to an address `recipient`, we can use the following API:
+
+Once an object is owned by another object, it is required that any references to that object go through its owner: If an object-owned object is passed as an input to an entry function in a transaction, that transaction will fail. For instance, if we have a chain of ownership: address `Addr1` owns object `a`, object `a` owns object `b`, and `b` owns object `c` then in order to use object `c` in a Move call, the transaction must be signed by `Addr1`, and accept `a` as an input, and `b` and `c` must be accessed dynamically during transaction execution:
```

Comment:
```suggestion
Once an object is contained as a dynamic field, it can only be accessed through its owner through the `dynamic_object_field` APIs. If an object-owned object is passed as an input to an entry function in a transaction, that transaction will fail. For instance, if we have a chain of ownership: address `Addr1` owns object `a`, object `a` owns object `b`, and `b` owns object `c` then in order to use object `c` in a Move call, the transaction must be signed by `Addr1`, and accept `a` as an input, and `b` and `c` must be accessed dynamically during transaction execution:
```